src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.ConnectionPools.cs

```diff
@@ -84,10 +90,9 @@ public void Dispose()
                 {
                     _isDisposed = true;
 
-                    // let all connections in the pool to go away
                     foreach (var (_, queue) in _pools)
                     {
-                        while (queue.TryDequeue(out var connection))
+                        while (queue.TryAcquire(out var connection))
```

Comment:
having a method called TryAcquire on a queue is actually less clear to me than calling it TrDequeue.  The latter makes it very clear it's being take out of the queue, the former does not.

Reply:
The var should be called pool.